math/src/main/java/com/powsybl/math/graph/UndirectedGraphImpl.java

```diff
@@ -572,21 +573,53 @@ public boolean traverse(int[] startingVertices, TraversalType traversalType, Tra
         return true;
     }
 
+    /**
+     * This method is used to get the size of a path considering only the elements validating the predicate
+     * @param path Path to size
+     * @param sizingPredicate Predicate used to filter the elements of the path
+     * @return size of the filtered path
+     */
+    private int filteredSize(TIntArrayList path, Predicate<? super E> sizingPredicate) {
```

Comment:
this could be defined by caller and simplified as the predicate is always the same. And you could maybe use TIntArrayList::grep to have something more compact, instead of transforming arrayList to array to stream.